nservicebus/messaging/messages-as-interfaces.md

```diff
@@ -1,21 +1,97 @@
 ---
 title: Messages as Interfaces
-summary: NServiceBus supports messages being defined as interfaces
+summary: Define messages as interfaces in NServiceBus to support dynamic dispatch and polymorphic routing scenarios
 component: core
-reviewed: 2024-01-23
+reviewed: 2024-03-28
 redirects:
 - nservicebus/messages-as-interfaces
 ---
 
-NServiceBus supports using interfaces as messages to allow multiple inheritance, where one message can inherit from varieties of other messages. This is useful for solving certain message evolution scenarios.
+NServiceBus supports using interfaces as messages to allow dynamic dispatch and polymorphic message routing scenarios. This makes systems more flexible and maintainable by encouraging messages and handlers to be decoupled from each other. It's useful for evolving systems to allow adding new features without affecting existing functionality. It's best described using an example.
 
-Imagine that the business logic represents a state machine with states X and Y. When the system gets into state X, it publishes the message `EnteredStateX`. When the system gets into state Y, it publishes the message `EnteredStateY`. (For more information on how to publish a message, see below.)
+## Using NServiceBus for dynamic dispatch
 
-In the next version of the system, a new state Z is added, which represents the co-existence of both X and Y states. This can be achieved by defining a message `EnteredStateZ` which inherits from `EnteredStateX` and `EnteredStateY`.
+Consider the following handlers stub for a `UserCreated` message which inherits from the `IEvent` stub interface:
 
-When the system publishes the `EnteredStateZ` event, the clients that are subscribed to either `EnteredStateX` or `EnteredStateY` are notified.
+```csharp
```

Comment:
The code should be moved to snippets but I've left it as inline until this is reviewed